src/web/Ligare/web/scaffolding/scaffolder.py

```diff
@@ -226,6 +226,15 @@ def _scaffold_directory(
         # because the module renderer uses the module
         # template directory as the root directory.
         if META_TEMPLATE_NAME in templates:
+            # make the list of all template in the module available
+            # to __meta__.j2 so it can process it if needed.
+            # the directory prefix `templates/` is stripped, and __meta__.j2
```

Comment:
Do you envision any issues tying this to a specific path? Is it possible to do this within Python a `.templates` module instead? Or is the filesystem stuff a vital part of the scaffolding process?

Reply:
so the scaffolding does something there that i think isn't what you're considering, but let's figure that out.
 
the short of it is that templates are not modules, nor are they python code. mostly the scaffolder templates is a directory tree of Jinja2 template files, and scaffolder.py is just finding and rendering them. with that in mind, what the scaffolder is responsible for needs to be considered.
 
the scaffolder renders different kinds of applications with different optional modules and other options. to accomplish that, the scaffolder can't just find all templates and render them - a set of rules and expectations needs to be created. some of those rules decide, for example, that the openapi endpoint application.py.j2 file overwrites the one that comes from the base templates. there are yet more rules for scaffolder modules - they have more complexity and are able to modify the behavior of the scaffolder while it's running. naturally, that complexity means even more rules and expectations.
 
that's what you're seeing here. when the scaffolder is looking for module templates, it looks for one called __meta__.j2, then makes certain data available to the Jinja2 template renderer so that __meta__.j2 can handle things specific to the module that it belongs to. for example, the database module meta renders alembic.ini from alembic.ini.j2 at the project root, rather than the default behavior that would render it in the module directory.
 
this specific code you're commenting on is the step immediately before __meta__.j2 is rendered. i decided that:
__meta__.j2 already knows that __meta__.j2 is in the list of templates being rendered for the module it is being called for, so remove it from the list of templates in the modules
with the exception of __meta__.j2 all templates exist in a templates/ directory because:
it's convenient for the scaffolder to render things under that directory only
ignoring the templates/ directory itself while rendering, allows the scaffolder to render templates without writing a templates/ directory
because all templates are under templates/ and this directory is intended to be ignored for rendering, it is removed so that __meta__.j2 can render them without messing with paths